cherrypick-2.0: log: log the cluster ID in new log files

knz opened the pull request:
Picks #24926.
cc @cockroachdb/release

knz committed:
log: log the cluster ID in new log files

The CONTRIBUTING guide suggest including the `[config]` lines with
error reports in new issues. This patch ensures these lines now
include the cluster ID. This is aimed to ease the correlation of
manual issue reports with auto-generated crash reports.

Release note (general change): The header of new log files generated
by `cockroach start` will now include the cluster ID once it has been
determined.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/24982)
<!-- Reviewable:end -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24982#-:-LAn6YS747oWyewkAJxF:bnfp4nl)*
<!-- Sent from Reviewable.io -->

knz committed:
util/log: ensure that also the first log file contains the cluster ID

Requested by @bdarnell. This ensures that `grep '\[config\]'
cockroach.log` finds it, even on the first log file.

Release note (general change): The cluster ID is also reported with
tag `[config]` in the first log file, not only when log files are
rotated.

knz commented:
I'll also pick #24993 here when that merges.

knz commented:
bors r+
